test/unit/view.js

```diff
@@ -5,143 +5,239 @@ import Iframe from '../../src/iframe';
+        assert.calledTwice(userEventStub);
+        assert.calledWith(userEventStub.getCall(0), 'beforeRender');
+        assert.calledWith(userEventStub.getCall(1), 'afterRender');
+        assert.calledOnce(templateRenderStub);
+      });
+
+      it('creates a wrapper if one does not already exist', () => {
+        view.wrapper = null;
+        view.render();
+        assert.calledOnce(createWrapperStub);
+        assert.equal(view.wrapper, div);
+      });
+
+      it('does not create a new wrapper if one already exists', () => {
+        view.wrapper = 'test';
+        view.render();
+        assert.notCalled(createWrapperStub);
+      });
+
+      it('updates node with wrapper and html then resizes', () => {
+        view.render();
+        assert.calledOnce(updateNodeStub);
+        assert.calledWith(updateNodeStub, view.wrapper, htmlTemplate);
+        assert.calledOnce(resizeStub);
       });
     });
 
     describe('delegateEvents()', () => {
-      let component;
-      let view;
+      let closeComponentsOnEscStub;
+      let onStub;
 
       beforeEach(() => {
-        component = new Component({
-          id: 1234,
-          node: document.createElement('div'),
-          options: {
-            product: {
-              DOMEvents: {
-                'click .btn'() { return true; },
-                'click'() { return true; },
-              },
-            },
+        component = Object.defineProperty(component, 'DOMEvents', {
+          value: {
+            'click .btn': sinon.spy(),
+            click: sinon.spy(),
           },
-        }, {browserFeatures: {}});
-        view = new View(component);
-        component.typeKey = 'product';
-      });
-
-      it('calls _on for each DOM event', () => {
-        const onStub = sinon.stub(view, '_on');
+        });
         view.wrapper = {
           addEventListener: sinon.spy(),
         };
+        closeComponentsOnEscStub = sinon.stub(view, 'closeComponentsOnEsc');
+        onStub = sinon.stub(view, '_on');
+      });
+
+      afterEach(() => {
+        closeComponentsOnEscStub.restore();
+        onStub.restore();
+      });
+
+      it('calls closeComponentsOnEsc', () => {
+        view.delegateEvents();
+        assert.calledOnce(closeComponentsOnEscStub);
+      });
+
+      it('calls _on for each DOM event with a selector', () => {
         view.delegateEvents();
-        assert.calledWith(onStub, 'click', '.btn', sinon.match.func);
-        assert.calledWith(view.wrapper.addEventListener, 'click', sinon.match.func);
+        assert.calledOnce(onStub);
+        assert.calledWith(onStub, 'click', '.btn');
+        onStub.getCall(0).args[2]();
+        assert.calledOnce(component.DOMEvents['click .btn']);
+      });
+
+      it('adds event listener for DOM event without a selector', () => {
+        view.delegateEvents();
+        assert.calledWith(view.wrapper.addEventListener, 'click');
+        view.wrapper.addEventListener.getCall(0).args[1]();
```

Comment:
Same as above, can we pass in an `evt`, and check calledWith for the DOMEvents click?